### PR TITLE
2739 feat/add eui analysis

### PR DESCRIPTION
**Pull request by Ryoken**
#### What's this PR do?
Adds a new analysis for EUI

#### How should this be manually tested?
Select one or more properties and run an analysis, then select "EUI".   For it to work, you need 12 months of consecutive electricity meter readings on the property.

#### What are the relevant tickets?
https://github.com/SEED-platform/seed/issues/2739

### CODE REVIEW

File: seed/tests/test_analysis_pipelines.py [hunk 1]
```diff
@@ -926,3 +939,101 @@ def test_build_better_input_returns_errors_if_state_missing_info(self):
         self.assertEqual(2, len(errors))
         self.assertTrue("BETTER analysis requires the property's name." in errors)
         self.assertTrue("BETTER analysis requires the property's city." in errors)
+
+
+class TestEuiPipeline(TestCase):
```
Comment (macintoshpie) [inline seed/tests/test_analysis_pipelines.py:944]: 🙌  Thank you for adding these!

File: seed/analysis_pipelines/eui.py [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+ERROR_NO_VALID_PROPERTIES = 3
+WARNING_SOME_INVALID_PROPERTIES = 4
+
+EUI_ANALYSIS_MESSAGES = {
+    ERROR_INVALID_GROSS_FLOOR_AREA: 'Property view skipped (invalid Gross Floor Area).',
+    ERROR_INSUFFICIENT_METER_READINGS: 'Property view skipped (no linked electricity meters with 12 or more readings).',
+    ERROR_INVALID_METER_READINGS: 'Property view skipped (no linked electricity meters with 12 months of consecutive readings).',
+    ERROR_NO_VALID_PROPERTIES: 'Analysis found no valid properties.',
+    WARNING_SOME_INVALID_PROPERTIES: 'Some properties failed to validate.'
+}
```
Comment (macintoshpie) [inline seed/analysis_pipelines/eui.py:43]: 😎  🔥

File: seed/analysis_pipelines/eui.py [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+    def _prepare_analysis(self, property_view_ids):
+        meter_readings_by_property_view, errors_by_property_view_id = _get_valid_meters(property_view_ids)
+
+        if not meter_readings_by_property_view:
+            AnalysisMessage.log_and_create(
+                logger=logger,
+                type_=AnalysisMessage.ERROR,
+                analysis_id=self._analysis_id,
+                analysis_property_view_id=None,
+                user_message=EUI_ANALYSIS_MESSAGES.ERROR_NO_VALID_PROPERTIES,
```
Comment (macintoshpie) [inline seed/analysis_pipelines/eui.py:144]:
```suggestion
                user_message=EUI_ANALYSIS_MESSAGES[ERROR_NO_VALID_PROPERTIES],
```

**Review by macintoshpie — CHANGES_REQUESTED** [review]
This is awesome, great work on the consecutive meter stuff.

One issue I had though was the case where all properties are invalid. No analysis property views are created, so when viewing the analysis you don't know exactly why the properties were "bad"
Screen Shot 2021-08-20 at 10 17 32 AM

Could you tweak the flow so that these analysis property views are still created and the messages get linked directly to them?

**Comment by macintoshpie** [comment]
> One issue I had though was the case where all properties are invalid. No analysis property views are created, so when viewing the analysis you don't know exactly why the properties were "bad"

On second thought let's just hold off on changing this too much. We should have a design discussion about how to document analyses and failure cases

File: seed/analysis_pipelines/eui.py (hunk 3 above)
Comment (Ryoken) [reply seed/analysis_pipelines/eui.py:144]: oops, good catch!

**Review by macintoshpie — APPROVED** [review]
🔥 🔥 🔥

**Pull request merged**